frontend/src/component/ui/Toggle.spec.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Toggle } from './Toggle';
+
+describe('Toggle component', () => {
+  test('should render without crash', () => {
+    render(<Toggle />);
+  });
+
+  test('should render an inactive toggle when disabled by default', () => {
+    const { container } = render(<Toggle defaultIsEnabled={false} />);
+
+    expect(container.querySelector('.background')).toBeTruthy();
+    expect(container.querySelector('.knob')).toBeTruthy();
+    expect(container.querySelector('.background.active')).toBeNull();
+    expect(container.querySelector('.knob.active')).toBeNull();
+  });
+
+  test('should render an active toggle when enabled by default', () => {
+    const { container } = render(<Toggle defaultIsEnabled={true} />);
+
+    expect(container.querySelector('.background.active')).toBeTruthy();
+    expect(container.querySelector('.knob.active')).toBeTruthy();
+  });
+
+  test('should toggle from disabled to enabled', () => {
+    const toggleRef = React.createRef<Toggle>();
+    const { container } = render(
+      <Toggle ref={toggleRef} defaultIsEnabled={false} />
+    );
+
```

Comment:
Please check the previous status to make sure it's actually disabled and the click actually enabled the toggle.

Reply:
Done